MainMenuSample/Source/Game/ButtonFuctions.cs

```diff
@@ -0,0 +1,306 @@
+            OptionsBackButton.Get<Button>().ButtonClicked -= ToggleOptionsMenu;
+            FullscreenButton.Get<Button>().ButtonClicked -= ToggleFullscreen;
+            GraphicsSettingButton.Get<Button>().ButtonClicked -= ModifyGraphicsSettingQuality;
+            AudioButton.Get<Button>().ButtonClicked -= ToggleAudio;
+            QuitButton.Get<Button>().ButtonClicked -= QuitButtonFunction;
+        }
+
+        public string[] ReadSettingsFile()
+        {
+            _lastSettingsReadValue = File.ReadAllLines(_dataPath);
+            return _lastSettingsReadValue;
+        }
+
+        public void WriteSettingsFile(bool GraphicsStatus, bool FullscreenStatus, bool AudioStatus)
+        {
+            string _writtenSettingsValue;
+            string _retreivedSettings;
+
+            if (GraphicsStatus)
+            {
+                _retreivedSettings = GFX_Preset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = GFX_Preset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+            if (FullscreenStatus)
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + FullscreenPreset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + FullscreenPreset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+            if (AudioStatus)
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + SFX_Preset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + SFX_Preset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+
+
+            if (EnableSettingsFileComments)
+            {
+                string _settingsComment = "// These values represent the in-game settings." +
+                 Environment.NewLine + "// The values match as following:" + Environment.NewLine + "// Setting enabled -> " + SettingsFileTrueTerm + 
+                 Environment.NewLine + "// Setting disabled -> " + SettingsFileFalseTerm + Environment.NewLine + "// Non-matching values will result in 'False' value.";
+
+                _writtenSettingsValue = _retreivedSettings + Environment.NewLine + Environment.NewLine + _settingsComment; // This is where the 2 bonus newlines come from.
+            }else
+            {
+                _writtenSettingsValue = _retreivedSettings;
+            }
+
+            File.WriteAllText(_dataPath, _writtenSettingsValue);
+        }
+
+        public void PlayButtonFunction(Button button)
+        {
+            Level.ChangeSceneAsync(GameScene);
+        }
+
+        public void ToggleOptionsMenu(Button button)
+        {
+            if (!_optionsMenuActive)
+            {
+                FullscreenButton.IsActive = true;

```

Comment:
Wouldn't it be easier to put all those buttons into a panel and enable/disable that panel?